services/tables/src/main/java/com/linkedin/openhouse/tables/api/validator/impl/OpenHouseTablesApiValidator.java

```diff
@@ -318,6 +318,31 @@ public void validateCreateLock(
     }
   }
 
+  @Override
+  public void validateSearchSoftDeletedTables(
+      String databaseId, String tableId, int page, int size, String sortBy) {
+    List<String> validationFailures = new ArrayList<>();
+    validateDatabaseId(databaseId, validationFailures);
+    // TableId is optional so can be null
+    if (tableId != null) {
+      validateTableId(tableId, validationFailures);
+    }
+    ApiValidatorUtil.validatePageable(page, size, sortBy, validationFailures);
+  }
+
+  @Override
+  public void validatePurgeSoftDeletedTable(String databaseId, String tableId, long purgeAfterMs) {
+    if (databaseId == null || tableId == null) {
+      throw new RequestValidationFailureException(
+          "databaseId and tableId must be provided to purge a soft deleted table");
+    }
+
+    if (purgeAfterMs < 0) {
```

Comment:
Should we also have ttl duration validation for purgeAfterMs? For example, diff between purgeAfterMs and deletedAt should not be greater than 7 days.

Reply:
Yeah let's not validate on that until user configurable TTL is set